server/resourcemanager/resource_manager.go

```diff
@@ -116,7 +116,7 @@ func (m *manager[T]) RegisterRoutes(r *mux.Router) *mux.Router {
 		subrouter.HandleFunc("/", m.create).Methods(http.MethodPost).Name(fmt.Sprintf("%s.Create", m.resourceType))
 	}
 	if slices.Contains(enabledOps, OperationUpdate) {
-		subrouter.HandleFunc("/", m.update).Methods(http.MethodPut).Name(fmt.Sprintf("%s.Update", m.resourceType))
+		subrouter.HandleFunc("/{id}", m.update).Methods(http.MethodPut).Name(fmt.Sprintf("%s.Update", m.resourceType))
```

Comment:
To be RESTful, our update API should receive an ID since the update operation should be made into one resource instead of one collection.
Source: https://cloud.google.com/apis/design/standard_methods